include/eve/algo/container/detail/soa_storage2.hpp

```diff
@@ -0,0 +1,172 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/module/core.hpp>
+#include <eve/algo/views/zip.hpp>
+#include <eve/detail/kumi.hpp>
+#include <eve/detail/raberu.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/product_type.hpp>
+#include <memory>
+#include <type_traits>
+
+namespace eve::algo::detail
+{
+  template<eve::product_type Type, typename Allocator>
+  struct soa_storage : Allocator
+  {
+    using is_product_type = void;
+
+    soa_storage()                : storage_( sizing(0) ), size_(0) { build(0); }
+    soa_storage( std::size_t n)  : storage_( sizing(n) ), size_(n) { build(n); }
```

Comment:
Why not just call the n, v one?
explicit please

Reply:
It was written before and not updated, will do.
Adding explicit too.